### PR TITLE
Fixes to browser detection

### PR DESCRIPTION
**Pull request by thomasdenney**
Per https://github.com/Microsoft/pxt/issues/355, Safari throws an exception when trying to access local storage in private browsing mode. This pull request therefore prevents access to the editor when local storage is not available, as well as a few other minor corrections to the browser check.

### CODE REVIEW

**Comment by pelikhan** [comment]
If local storage is unavailable, we need to use an in memory version.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by thomasdenney** [comment]
OK - the changes to the browser detection code (Safari version, navigator check) should be separated out, as these are bug fixes that should be merged.

**Comment by thomasdenney** [comment]
I've now removed the requirement for local storage in the browser check, but this pull request does include other fixes to the browser detection code (Safari version number). I've also left the function that checks whether local storage is available, as this will be useful for determining whether to use local storage or the in memory workspace.

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -1629,8 +1629,26 @@ function initHashchange() {
     });
 }
 
+function unsupportedBrowserRedirect() {
+    Util.httpGetJsonAsync(pxt.webConfig.targetCdnUrl + "target.json")
```
Comment (pelikhan) [inline webapp/src/app.tsx]: why are we reloading the target.json here?

File: pxtlib/browserutils.ts [hunk 2]
```diff
@@ -135,8 +135,20 @@ namespace pxt.BrowserUtils {
 
     let hasLoggedBrowser = false
 
+    export function isLocalStorageSupported(): boolean {
```
Comment (pelikhan) [inline pxtlib/browserutils.ts]: This helper should be used in pxt.storage to test if localStorage can be used. Should be moved in new PR.

File: pxtlib/browserutils.ts [hunk 3]
```diff
@@ -113,7 +113,7 @@ namespace pxt.BrowserUtils {
             matches = /Midori\/([0-9\.]+)/i.exec(navigator.userAgent);
         }
         else if (isSafari()) {
-            matches = /Safari\/([0-9\.]+)/i.exec(navigator.userAgent);
+            matches = /Version\/([0-9\.]+)/i.exec(navigator.userAgent);
```
Comment (pelikhan) [inline pxtlib/browserutils.ts]: This fix could be bundled in separate PR.

**Comment by thomasdenney** [comment]
@pelikhan I'm going to separate this out into two separate pull requests for the updated browser checking code and the function for checking the availability of local storage.
